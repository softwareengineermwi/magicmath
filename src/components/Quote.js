import React from 'react';

const Quote = () => (
  <div className="container text-component padding-y-lg" data-testid="quote">
    <h4>
      Mathematics is not about numbers,equations,computations,or algorithms:
      its about understanding.
      <cite>- William Paul Thurston.</cite>
    </h4>
  </div>
);

export default Quote;
