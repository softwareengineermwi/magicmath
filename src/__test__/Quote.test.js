import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Quote from '../components/Quote';
import '@testing-library/jest-dom';

describe('Test Quote Components', () => {
  it('Should render without error', () => {
    const divElement = document.createElement('div');
    ReactDOM.render(<Quote />, divElement);
  });

  it('Should render correctly', () => {
    const { getByTestId } = render(<Quote />);
    expect(getByTestId('quote')).toHaveTextContent('Mathematics is not about numbers,equations,computations,or algorithms: its about understanding.- William Paul Thurston.');
  });

  it('Should match snapshot', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
