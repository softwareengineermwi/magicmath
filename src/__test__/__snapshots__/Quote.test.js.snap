// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Quote Components Should match snapshot 1`] = `
<div
  className="container text-component padding-y-lg"
  data-testid="quote"
>
  <h4>
    Mathematics is not about numbers,equations,computations,or algorithms: its about understanding.
    <cite>
      - William Paul Thurston.
    </cite>
  </h4>
</div>
`;
